book-content/chapters/11-annotations-and-assertions.md

```diff
@@ -0,0 +1,1196 @@
+When there's a choice with how to suppress an error, I prefer using `as any`. Error suppression directives are too broad - they target the entire line of code. This can lead to accidentally suppressing errors that you didn't mean to:
+
+```typescript
+// @ts-ignore
+const result = addone("one");
+```
+
+Here, we're calling `addone` instead of `addOne`. The error suppression directive will suppress the error, but it will also suppress any other errors that might occur on that line.
+
+Using `as any` instead is more precise:
+
+```ts twoslash
+// @errors: 2552
+const addOne = (num: number) => num + 1;
+// ---cut---
+const result = addone("one" as any);
+```
+
+Now, you'll only suppress the error that you intended to.
+
+## When To Suppress Errors
+
+Each of the error suppression tools we've looked at is a way of basically telling TypeScript to "keep quiet". TypeScript doesn't attempt to limit how often you try to silence it. It's perfectly possible that every time you encounter an error, you could suppress it with `@ts-ignore` or `as any`.
+
+Taking this approach limits how useful TypeScript can be. Your code will compile, but you will likely get many more runtime errors.
+
+But there are times when suppressing errors is a good idea. Let's explore a few different scenarios.
+
+### When You Know More Than TypeScript
+
+The important thing to remember about TypeScript is that really, you're writing JavaScript.
+
+This disconnect between compile time and runtime means that types _can sometimes be wrong_. This can mean you know more about the runtime code than TypeScript does.
+
+This can happen when third-party libraries don't have good type definitions, or when you're working with a complex pattern that TypeScript struggles to understand.
+
+Error suppression directives exist for this reason. They let you patch over the differences that sometimes crop up between TypeScript and the JavaScript it produces.
+
+But this feeling of superiority over TypeScript can be dangerous. So, let's compare it to a very similar feeling:
+
+### When TypeScript Is Being "Dumb"
+
+Some patterns lend themselves better to being typed than others. More dynamic patterns can be harder for TypeScript to understand, and will lead you to suppressing more errors.
+
+A simple example is constructing an object. In JavaScript, there's no real difference between these two patterns:
+
+```ts twoslash
+// @errors: 2339
+// Static
+const obj = {
+  a: 1,
+  b: 2,
+};
+
+// Dynamic
+const obj2 = {};
+
+obj2.a = 1;
+obj2.b = 2;
+```
+
+In the first, we construct an object by passing in the keys and values. In the second, we construct an empty object and add the keys and values later. The first pattern is static, the second is dynamic.
+
+But in TypeScript, the first pattern is much easier to work with. TypeScript can infer the type of `obj` as `{ a: number, b: number }`. But it can't infer the type of `obj2` - it's just an empty object. In fact, you'll get errors when you try to do this.
```

Comment:
Fun fact, TS supports the dynamic variant in JS files :D it doesn't do it in overly safe way though:
```ts
const obj = {}
obj.a = 1

if (Math.random()) {
    obj.b = 'foo'
}

obj // { a: number; b: string }
```